mmeval/metrics/keypoint_nme.py

```diff
@@ -0,0 +1,261 @@
+
+    Calculate the normalized mean error (NME) of keypoints.
+
+    Note:
+        - length of dataset: N
+        - num_keypoints: K
+        - number of keypoint dimensions: D (typically D = 2)
+
+    Args:
+        norm_mode (str): The normalization mode. There are two valid modes:
+            `'use_norm_item'` and `'keypoint_distance'`.
+            When set as `'use_norm_item'`, should specify the argument
+            `norm_item`, which represents the item in the datainfo that
+            will be used as the normalization factor.
+            When set as `'keypoint_distance'`, should specify the argument
+            `keypoint_indices` that are used to calculate the keypoint
+            distance as the normalization factor.
+        norm_item (str, optional): The item used as the normalization factor.
+            For example, `'box_size'` in `'AFLWDataset'`. Only valid when
+            ``norm_mode`` is ``use_norm_item``.
+            Defaults to ``None``.
+        keypoint_indices (Sequence[int], optional): The keypoint indices used
+            to calculate the keypoint distance as the normalization factor.
+            Only valid when ``norm_mode`` is ``keypoint_distance``.
+            If set as None, will use the default ``keypoint_indices`` in
+            `DEFAULT_KEYPOINT_INDICES` for specific datasets, else use the
+            given ``keypoint_indices`` of the dataset. Defaults to ``None``.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+
+        >>> from mmeval import KeypointNME
+        >>> import numpy as np
+        >>> aflw_dataset_meta = {
+        ...     'dataset_name': 'aflw',
+        ...     'num_keypoints': 19,
+        ...     'sigmas': np.array([]),
+        ... }
+        >>> def _generate_data(self,
+        ...                    batch_size: int = 1,
+        ...                    num_keypoints: int = 5,
+        ...                    norm_item: str = 'box_size'):
+        ...     predictions = []
+        ...     groundtruths = []
+        ...     for i in range(batch_size):
+        ...         keypoints = np.zeros((1, num_keypoints, 2))
+        ...         keypoints[0, i] = [0.5 * i, 0.5 * i]
+        ...         keypoints_visible = np.ones(
+        ...             (1, num_keypoints)).astype(bool)
+        ...         keypoints_visible[0, (2 * i) % 8] = False
+        ...         prediction = {'coords': keypoints}
+        ...         groundtruth = {
+        ...             'coords': keypoints,
+        ...             'mask': keypoints_visible,
+        ...             norm_item: np.random.random((1, 1)) * 20 * i
+        ...         }
+        ...         predictions.append(prediction)
+        ...         groundtruths.append(groundtruth)
+        ...     return predictions, groundtruths
+        >>> norm_item = 'box_size'
+        >>> predictions, groundtruths = _generate_data(
+        ...     batch_size=4, num_keypoints=19, norm_item=norm_item)
```

Comment:
It's better to simplify the example. I'm not familiar with this metric, but something like:
```python
batch_size = 2
predictions = [{'coords': np.zeros(1, 19, 2)} for _ in range(batch_size)]
groundtruths = [{
    'coords': np.zeros(1, 19, 2)+0.5,
    'mask': np.ones(1, 19).astype(bool),
    'box_size': np.ones((1, 1)) * i}
    for i in range(batch_size)
]
```
should be more easy to understand ^_^
Note: I don't know if the above example has proper data range, e.g. in `box_size`